presto-hive-metastore/src/main/java/com/facebook/presto/hive/metastore/thrift/ThriftHiveMetastore.java

```diff
@@ -1000,19 +1010,40 @@ private void addPartitionsWithoutStatistics(String databaseName, String tableNam
         }
     }
 
+    private <V> V getMetastoreClientThenCall(MetastoreContext metastoreContext, MetastoreCallable<V> callable)
+            throws Exception
+    {
+        if (!impersonationEnabled) {
+            HiveMetastoreClient client = clientProvider.createMetastoreClient(Optional.empty());
+            return callable.call(client, metastoreHostStats.get(client.getAddress()));
+        }
+        try (HiveMetastoreClient client = clientProvider.createMetastoreClient(Optional.empty())) {
```

Comment:
Why do we need to keep the `HiveMetastoreClient client` open for the entire period of the underlying call? Can it be closed right after retrieving the delegation token?